test/model/internal_link_test.dart

```diff
@@ -221,6 +221,16 @@ void main() {
       testExpectedNarrows(testCases, streams: streams);
     });
 
+    group('"/#narrow/is/mentioned returns expected MentionsNarrow', () {
+      final testCases = [
+        ('/#narrow/is/mentioned',        const MentionsNarrow()),
+        ('/#narrow/is/mentioned/near/1', const MentionsNarrow()),
+        ('/#narrow/is/mentioned/with/2', const MentionsNarrow()),
+        ('/#narrow/-is/mentioned',       null),
+      ];
```

Comment:
Let's add cases like:

```dart
('/#narrow/stream/7-test-here/is/mentioned', null),
('/#narrow/dm/17327-Chris-Bobbe-(Test-Account)/is/mentioned', null),
```

Those are valid narrows to want to see in a client, but we don't support them yet, and it seems good that we're taking the user out to the browser for those, rather than showing the whole `/is/mentioned` narrow in the app.

Reply:
Also added one with the topic segment. Fixed a bug with these new cases!